### ISSUE TITLE
How to load ivy dependencies through a HTTP proxy

### ISSUE THREAD

**Issue by DavidPerezIngeniero** [lihaoyi/Ammonite#460]
In an Ammonite script I have this line:

```
import $ivy.`org.postgresql:postgresql:9.4.1209.jre7`
```

The result when running this script is:

```
Server access error at url http://repo1.maven.org/maven2/org/postgresql/postgresql/9.4.1209.jre7/postgresql-9.4.1209.jre7.pom (java.net.SocketException: Connection reset)
Exception in thread "main" ammonite.runtime.tools.IvyThing$IvyResolutionException: failed to resolve ivy dependencies unresolved dependency: org.postgresql#postgresql;9.4.1209.jre7: not found
```

It seems that a proxy isn't used even though the environment variables `http_proxy` and `https_proxy` are properly set.

**Comment by lihaoyi** [comment]
Ammonite doesn't provide any way to allow proxies right now; would be happy to make it take them as something you could initialize in your predef, like how we do resolvers today

**Comment by DavidPerezIngeniero** [comment]
Or like SBT that autodetects proxies by using the http_proxy environment variable. (At least in Linux).

**Comment by PerArneng** [comment]
This would be a very nice feature to have. It is very hard to make use of the excellent Ammonite when behind a corporate firewall with a mandatory http proxy.

**Comment by PerArneng** [comment]
It seems to use the proxy when these are set.
```
$ export JAVA_OPTS="-Dhttp.proxyHost=my.little.proxy.com -Dhttp.proxyPort=808"
```
However it adds the scala version to all the dependencies and can not find them.

```
$ ./test.sc
Compiling test.sc
:: loading settings :: url = jar:file:/usr/local/bin/amm!/org/apache/ivy/core/settings/ivysettings.xml
:: resolving dependencies :: org.fusesource.jansi#jansi_2.12-caller;working
        confs: [default]

:: problems summary ::
:::: WARNINGS
                module not found: org.fusesource.jansi#jansi_2.12;1.14

        ==== local: tried

          /home/kod425/.ivy2/local/org.fusesource.jansi/jansi_2.12/1.14/ivys/ivy.xml

          -- artifact org.fusesource.jansi#jansi_2.12;1.14!jansi_2.12.jar:

          /home/kod425/.ivy2/local/org.fusesource.jansi/jansi_2.12/1.14/jars/jansi_2.12.jar

        ==== m2: tried

          /home/kod425/.m2/repository/org/fusesource/jansi/jansi_2.12/1.14/ivy-1.14.xml

          -- artifact org.fusesource.jansi#jansi_2.12;1.14!jansi_2.12.jar:

          /home/kod425/.m2/repository/org/fusesource/jansi/jansi_2.12/1.14/jansi_2.12-1.14.jar

        ==== central: tried

          http://repo1.maven.org/maven2/org/fusesource/jansi/jansi_2.12/1.14/jansi_2.12-1.14.pom

          -- artifact org.fusesource.jansi#jansi_2.12;1.14!jansi_2.12.jar:

          http://repo1.maven.org/maven2/org/fusesource/jansi/jansi_2.12/1.14/jansi_2.12-1.14.jar


:: USE VERBOSE OR DEBUG MESSAGE LEVEL FOR MORE DETAILS
Exception in thread "main" ammonite.runtime.tools.IvyThing$IvyResolutionException: failed to resolve ivy dependencies unresolved dependency: org.fusesource.jansi#jansi_2.12;1.14: not found

```

*The script*
```scala
#!/usr/bin/env amm

import $ivy.`org.fusesource.jansi::jansi:1.14`
import org.fusesource.jansi.AnsiConsole

AnsiConsole.systemInstall();
AnsiConsole.out.println("Hello World");
```

**Comment by cuzfrog** [comment]
I searched ivy api, turns out ivy reads jvm system properties for proxy settings.
Simply adding to top of `predef.sc` with system properties solved my problem:
```scala
System.setProperty("http.proxyHost","some_address")
System.setProperty("http.proxyPort","some_port")
//https etc.
```

However, this solution is not very neat.  Placing a `config.properties` to `.ammonite/` along side with `predef.sc` with some default definition in `Storage` might be better?

Either proxy solution deserves lines in the doc, it would save others some time. Thanks.

**Comment by DavidPerezIngeniero** [comment]
You could make branch the documentation and make a feature request

**Comment by cuzfrog** [comment]
I made a PR, how to make feature request? By the way, this enhancement is somewhat intuitive and not worth a feature, I think.

**Comment by lihaoyi** [comment]
@alexarchambault does Coursier have something special for http proxying?

**Comment by alexarchambault** [comment]
As pointed out on https://github.com/lihaoyi/Ammonite/pull/563, the setup described in https://docs.oracle.com/javase/7/docs/api/java/net/doc-files/net-properties.html works with coursier (as coursier relies on standard `java.net.HttpUrlConnection`).

**Comment by lihaoyi** [comment]
Should be fixed in #563
